src/filesystem.tsx

```diff
@@ -57,14 +57,16 @@ export async function readFile(file: FileEntry) {
 function sortedFileEntries(entries: FileEntry[]): FileEntry[] {
   return entries
     .sort((fileA, fileB) => {
-      if (fileA.children === null) return -1;
-      if (fileB.children === null) return 1;
-      return fileA.name?.localeCompare(fileB.name || '') || 0;
+      if (fileA.children === undefined) {
```

Comment:
This is [@typescript-eslint/no-unnecessary-condition](https://typescript-eslint.io/rules/no-unnecessary-condition) and seems to be a real issue (the comparison should be against `undefined`, not `null`).